src/kubeobjects/address/address.go

```diff
@@ -1,15 +1,5 @@
 package address
 
-import (
-	"time"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
-
-type scalarType interface {
-	bool | int | int64 | time.Time | metav1.Time
-}
-
-func Of[T scalarType](i T) *T {
-	return &i
+func Of[A any](value A) *A {
```

Comment:
revert to restrict function to scalar types.

this is intended as a helper to initialize pointers to scalar types, e.g. integers (`address.Of(1)`), not for general structs, which can be initialized as pointers directly via `&`.

Reply:
Needed such an idiom to cope with the function result. To not propagate the pattern with the address operator on the transient variable.

Is not the given construct verbose?
```
func NewQuantity(serialized string) *resource.Quantity {
	quantity := resource.MustParse(serialized)
	return &quantity
}
```

Reply:
Quoted Marcell's phrase.